lib/events/athena/publisher.go

```diff
@@ -94,11 +158,28 @@ func newPublisherFromAthenaConfig(cfg Config) *publisher {
 		so.MaxAttempts = 20
 		so.MaxBackoff = 1 * time.Minute
 	})
-	return NewPublisher(PublisherConfig{
-		TopicARN: cfg.TopicARN,
-		SNSPublisher: sns.NewFromConfig(*cfg.PublisherConsumerAWSConfig, func(o *sns.Options) {
+	hc := awshttp.NewBuildableClient().WithTransportOptions(func(t *http.Transport) {
+		// attempt to aggressively reuse connections for the sake of
+		// avoiding TLS handshakes (the default MaxIdleConnsPerHost is a
+		// pitiful 2)
+		t.MaxIdleConns = 10000
+		t.MaxIdleConnsPerHost = 10000
+	})
+	var messagePublisher messagePublisherFunc
+	if cfg.TopicARN == "" {
```

Comment:
If it is easy to align - can we add at least 1 test in [integraton_test.go](https://github.com/gravitational/teleport/blob/b178b8b7322e7c6348bbd17ed1d3251ad34d946b/lib/events/athena/integration_test.go#L42) for `cfg.TopicARN == ""` case ?

Reply:
I couldn't find a sensible way without copying a bunch of code so I just added an envvar to bypass SNS when running tests. 😬